aliyun/src/test/java/org/apache/iceberg/aliyun/oss/TestOSSOutputFile.java

```diff
@@ -0,0 +1,137 @@
+  private final Random random = ThreadLocalRandom.current();
+  private final AliyunProperties aliyunProperties = new AliyunProperties();
+
+  @Test
+  public void testWriteFile() throws IOException {
+    OSSURI uri = randomURI();
+
+    int dataSize = 1024 * 1024 * 10;
+    byte[] data = randomData(dataSize);
+
+    OutputFile out = OSSOutputFile.fromLocation(ossClient, uri.location(), aliyunProperties);
+    try (OutputStream os = out.createOrOverwrite()) {
+      IOUtils.write(data, os);
+    }
+    InputFile in = out.toInputFile();
+    Assert.assertTrue("OSS file should exist", in.exists());
+
+    byte[] actual = new byte[dataSize];
+    ByteStreams.readFully(ossClient.getObject(uri.bucket(), uri.key()).getObjectContent(), actual);
+    Assert.assertArrayEquals("Object content should match", data, actual);
+  }
+
+  @Test
+  public void testFromLocation() {
+    AssertHelpers.assertThrows(
+        "Should catch null location when creating oss output file",
+        NullPointerException.class,
+        "location cannot be null",
+        () -> OSSOutputFile.fromLocation(ossClient, null, aliyunProperties));
+  }
+
+  @Test
+  public void testCreate() {
+    OSSURI uri = randomURI();
+
+    int dataSize = 8;
+    byte[] data = randomData(dataSize);
+    writeOSSData(uri, data);
+
+    OutputFile out = OSSOutputFile.fromLocation(ossClient, uri.location(), aliyunProperties);
+    AssertHelpers.assertThrows("Should complain about location already exists", AlreadyExistsException.class,
+        "Location already exists",
+        () -> out.create());
+  }
+
+  @Test
+  public void testCreateOrOverwrite() throws Exception {
+    OSSURI uri = randomURI();
+
+    int dataSize = 8;
+    byte[] data = randomData(dataSize);
+    writeOSSData(uri, data);
+
+    int actualSize = 1024;
+    byte[] actual = randomData(actualSize);
+    OutputFile out = OSSOutputFile.fromLocation(ossClient, uri.location(), aliyunProperties);
+    try (OutputStream os = out.createOrOverwrite()) {
+      IOUtils.write(actual, os);
+    }
+
+    Assert.assertEquals(String.format("Should overwrite object length from %d to %d", dataSize, actualSize),
+        ossClient.getObject(uri.bucket(), uri.key()).getObjectMetadata().getContentLength(), actualSize);
+
+    byte[] expect = new byte[actualSize];
+    ByteStreams.readFully(ossClient.getObject(uri.bucket(), uri.key()).getObjectContent(), expect);
+    Assert.assertArrayEquals("Should overwrite object content", actual, expect);
+  }
+
+  @Test
+  public void testLocation() {
+    OSSURI uri = randomURI();
+    OutputFile out = new OSSOutputFile(ossClient, uri, aliyunProperties);
+    Assert.assertEquals("Location should match", uri.location(), out.location());
+  }
+
+  @Test
+  public void testToInputFile() {
+    OutputFile out = new OSSOutputFile(ossClient, randomURI(), aliyunProperties);
+    InputFile in = out.toInputFile();
+    Assert.assertTrue("Should be an instance of OSSInputFile", in instanceof OSSInputFile);
```

Comment:
I think we'd better to assert more informations such as: 
* the location of output file and input file;
* the length of the output file and input file;
* the existence of the input file.